packages/core/src/account/base.ts

```diff
@@ -48,7 +48,7 @@ export abstract class BaseSmartContractAccount<
     Chain
   >;
   protected entryPointAddress: Address;
-  protected rpcProvider:
+  readonly rpcProvider:
```

Comment:
we might want to expose this concept on `ISmartContractAccount` or at least via a `getRpcProvider` for `ISmartContractAccount`